shadowsocks-csharp/Controller/Service/TCPRelay.cs

```diff
@@ -586,39 +490,58 @@ private void PipeRemoteReceiveCallback(IAsyncResult ar)
 
         private void PipeConnectionReceiveCallback(IAsyncResult ar)
         {
-            if (closed)
-            {
-                return;
-            }
+            if (_closed) return;
             try
             {
                 int bytesRead = connection.EndReceive(ar);
-                totalWrite += bytesRead;
-
+                _totalWrite += bytesRead;
                 if (bytesRead > 0)
                 {
+                    int atyp = _connetionRecvBuffer[0];
+                    string dst_addr;
+                    int dst_port;
+                    switch (atyp)
+                    {
+                        case 1:  // IPv4 address, 4 bytes
+                            dst_addr = new IPAddress(_connetionRecvBuffer.Skip(1).Take(4).ToArray()).ToString();
+                            dst_port = (_connetionRecvBuffer[5] << 8) + _connetionRecvBuffer[6];
+                            if ( controller.GetCurrentConfiguration().isVerboseLogging ) {

```

Comment:
will refine "controller.GetCurrentConfiguration()" after PR #604 get merged, then we can switch to something like "config.isVerboseLogging"